core/shared/src/main/scala/zio/stm/TMap.scala

```diff
@@ -287,24 +287,19 @@ final class TMap[K, V] private (
       val buckets  = tBuckets.unsafeGet(journal)
       val capacity = buckets.array.length
       val size     = tSize.unsafeGet(journal)
-      val data     = Array.ofDim[(K, V)](size)
       var i        = 0
-      var j        = 0
+      val builder  = ChunkBuilder.make[(K, V)](size)
 
       while (i < capacity) {
         val bucket = buckets.array(i)
         val pairs  = bucket.unsafeGet(journal)
 
-        val it = pairs.iterator
-        while (it.hasNext) {
-          data(j) = it.next()
-          j += 1
-        }
+        builder.addAll(pairs)
```

Comment:
One last thing, since cross-build is "yelling", this can be replaced with `++=`. So:

```scala
while (i < capacity) {
  val bucket = buckets.array(i)

  builder ++= bucket.unsafeGet(journal)

  i += 1
}
```